api/v1/constructors.go

```diff
@@ -1437,30 +1437,10 @@ func NewPlatformNetworkSpec(pool addresspools.AddressPool, network networks.Netw
 		allocation_type = networks.AllocationOrderStatic
 	}
 	spec := PlatformNetworkSpec{
-		Type:               network.Type,
-		Subnet:             pool.Network,
-		FloatingAddress:    pool.FloatingAddress,
-		Controller0Address: pool.Controller0Address,
-		Controller1Address: pool.Controller1Address,
-		Prefix:             pool.Prefix,
-		Gateway:            pool.Gateway,
-		Allocation: AllocationInfo{
-			Type:  allocation_type,
-			Order: &pool.Order,
-		},
+		NetworkType:      network.Type,
```

Comment:
Do we need to construct the new address pool spec?

Reply:
Added constructor for AddressPool spec.
Had kept it for deployctl commit but we now have it anyway, PlatformNetwork constructor will be updated during deployctl commit to support exporting associatedAddressPools though.